### PR TITLE
Refactor Windshaft client - Polygon Decoder

### PR DESCRIPTION
**Pull request by elenatorro**
This is the first change in the process of refactoring [client/windshaft.js](https://github.com/CartoDB/carto-vl/blob/master/src/client/windshaft.js) in order to organize it in different files.

The main idea is to move all the methods related with the decode process to another directory. After reading the documentation, I think it could be something like:

```
|- / src
|-- / client
|--- / mvt
|---- feature-decoder.js
|---- layer-decoder.js
|---- value-decoder.js
```
What do you think?

### CODE REVIEW

File: src/utils/geometry.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (elenatorro) [reply src/utils/geometry.js]: ouch!

File: src/utils/geometry.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    // Add polygon composed by preClippedVertices to the `polygon.flat` array
+    clipPolygon(preClippedVertices, polygon, isHole) {
```
Comment (IagoLast) [inline src/utils/geometry.js]: maybe we could move this method to the `Polygon` class as `clip `and do

```js
clipped = polygon.clip(precliped, ishole);
```

maybe with some more refactor we can do only

```js
clipped = polygon.clip();
```

File: src/utils/geometry.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+function clipPolygon(preClippedVertices, polygon, isHole) {
```
Comment (IagoLast) [inline src/utils/geometry.js]: not sure about the need of having this method inside a utility file I think `clipPolygon` is only used in `feature-decoder.js`

**Comment by IagoLast** [comment]
LGTM let @davidmanzanares say the final word

**Pull request merged**